backend/capellacollab/core/authentication/jwt_bearer.py

```diff
@@ -25,43 +24,50 @@ class JWTBearer(HTTPBearer):
     def __init__(self, auto_error: bool = True):
         super(JWTBearer, self).__init__(auto_error=auto_error)
 
-    async def __call__(self, request: Request):
-        credentials: HTTPAuthorizationCredentials = await super(
+    async def __call__(
+        self, request: Request
+    ) -> t.Optional[t.Dict[str, t.Any]]:
+        credentials: t.Optional[HTTPAuthorizationCredentials] = await super(
             JWTBearer, self
         ).__call__(request)
+
         if not credentials or credentials.scheme != "Bearer":
             if self.auto_error:
                 raise HTTPException(
                     status_code=401,
                     detail="Not authenticated",
                     headers={"WWW-Authenticate": "Bearer"},
                 )
-            else:
-                return None
-        token_decoded = self.validate_token(credentials.credentials)
-        self.initialize_user(token_decoded)
-        return token_decoded
+            return None
+        if token_decoded := self.validate_token(credentials.credentials):
```

Comment:
I think we added this here because we wanted to fix the typing of the `JWTBearer` class. The general reason/problem why we need the `None` case here is that we specify with `self.auto_error` whether we want to throw a `HTTPException` or not and we wanted to add the check to the `validate_token` as well. Therefore, in case of a `jwt` exception and `auto_error` is false, we need to return `None` inside of `validate_toke` and that is the point where the token can be None.